### PR TITLE
Vet clinic database: query multiple tables

### PR DESCRIPTION
**Pull request by Alfred-KInyua**
### Project requirements 

- Use primary key & foreign key mechanisms for joining tables.
- Query multiple tables.
- Prepare complex queries that answer analytical questions.

I was able to answer the following questions. With the screenshots below  

- What animals belong to Melody Pond?
- List of all animals that are pokemon (their type is Pokemon).
- List all owners and their animals, remember to include those that don't own any animal.
- How many animals are there per species?
- List all Digimon owned by Jennifer Orwell.
- List all animals owned by Dean Winchester that haven't tried to escape.
- Who owns the most animals?

AA

BB

CC

DD

EE

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -9,3 +9,46 @@ SELECT * FROM animals WHERE name NOT IN ('Gabumon');
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
 
 
+update animals set species = name where name like '%mon%'; 
+update animals set species = 'pokemon' where species is null;
```
Comment (CollinsTatang) [inline queries.sql:13]: - [ ] Please we should write the query to update the animal name in the `data.sql`.

File: data.sql [hunk 2]
```diff
@@ -5,3 +5,44 @@ insert into animals (name, date_of_birth, escape_attempts, neutered, weight_in_k
...
+
+
+
+insert into animals(name, date_of_birth, escape_attempts, neutered,weight_in_kg)values('Charmander','2-8-2020',0,false,-11);
+insert into animals(name, date_of_birth, escape_attempts, neutered,weight_in_kg)values('Plantmon','2021-11-15',2,true,-5.7);
+insert into animals(name, date_of_birth, escape_attempts, neutered,weight_in_kg)values('Squirtle','1993-4-2',3,false,-12.13);
+insert into animals(name, date_of_birth, escape_attempts, neutered,weight_in_kg)values('Angemon','2005-6-12',1,true,-45);
+insert into animals(name, date_of_birth, escape_attempts, neutered,weight_in_kg)values('Boarmon','2005-6-7',7,true,20.4);
+insert into animals(name, date_of_birth, escape_attempts, neutered,weight_in_kg)values('Blossom','1998-10-13',3,true,17);
+insert into animals(name, date_of_birth, escape_attempts, neutered,weight_in_kg)values('Ditto','2022-5-14',4,true,22);
```
Comment (CollinsTatang) [inline data.sql]: - [ ] Please remove all unnecessary code from the `data.sql` file

File: queries.sql [hunk 3]
```diff
@@ -9,3 +9,46 @@ SELECT * FROM animals WHERE name NOT IN ('Gabumon');
...
+commit transaction 
+
+
+select count (name) from animals;
+select * from animals where escape_attempts =0; 
+select avg(weight_in_kg) from animals;
+select name min(weight_in_kg) and max(weight_in_kg) from animals group by name;
+select neutered, max(escape_attempts) from animals group by neutered;
+select name, avg(escape_attempts) from animals where date_of_birth BETWEEN '1990-1-1' AND '2000-12-31' group by name;
+
```
Comment (CollinsTatang) [inline queries.sql]: - [ ] Please remove all unnecessary code from the `queries.sql` file

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @Alfred-KInyua, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by smunozmo — APPROVED** [review]
Hi @Alfred-KInyua, 👋 

## Highlights 👏 

- Excellent job making the suggested changes! 😎 
- All transactions are done properly, perfect! 🎉 
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!

Your project is complete!  There is nothing else to say other than... it's time to merge it! 🚀 
Very nice work, congratulations! 🎉 

# APPROVED! 💯  💯  💯 




Cheers and Happy coding! 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**